### PR TITLE
 Feat: 🚀 Prepared addon for Keda

### PR DESCRIPTION
**Pull request by ballalamit**

### CODE REVIEW

File: addons/keda/variables.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    tags                           = map(string)
+  })
+}
+
+variable "eks_cluster_name" {
+  type    = string
+  default = ""
+}
+
+variable "account_id" {
```
Comment (h1manshu98) [inline addons/keda/variables.tf:33]: This can be removed if not using `irsa_config = {}`

**Comment by nileshgadgi** [comment]
Update workflow to automate the keda readme file creation.

File: addons/keda/locals.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    skip_crds                  = try(var.keda_extra_configs.skip_crds, "false")
+    render_subchart_notes      = try(var.keda_extra_configs.render_subchart_notes, "true")
+    disable_openapi_validation = try(var.keda_extra_configs.disable_openapi_validation, "false")
+    wait                       = try(var.keda_extra_configs.wait, "true")
+    wait_for_jobs              = try(var.keda_extra_configs.wait_for_jobs, "false")
+    dependency_update          = try(var.keda_extra_configs.dependency_update, "false")
+    replace                    = try(var.keda_extra_configs.replace, "false")
+  }
+
+  keda_extra_configs = var.keda_extra_configs
```
Comment (h1manshu98) [inline addons/keda/locals.tf:37]: This can be removed, as we are not using `local.keda_extra_configs` anywhere

File: addons/keda/variables.tf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  default = ""
+}
+
+variable "keda_extra_configs" {
+  description = "Override attributes of helm_release terraform resource"
+  type        = any
+  default     = {}
+}
+
+variable "iampolicy_json_content" {
```
Comment (h1manshu98) [inline addons/keda/variables.tf:44]: Remove this if not using `irsa_config = {}`

File: addons/keda/locals.tf (hunk 2 above)
Comment (ballalamit) [reply addons/keda/locals.tf]: done

File: addons/keda/variables.tf (hunk 1 above)
Comment (ballalamit) [reply addons/keda/variables.tf]: done

File: addons/keda/variables.tf (hunk 3 above)
Comment (ballalamit) [reply addons/keda/variables.tf]: done

**Comment by nileshgadgi** [comment]
image
please remove this variable in kube-state-matrics

**Comment by ballalamit** [comment]
> image please remove this variable in kube-state-matrics

I have not worked on this addons, but I can remove this if gets the permission to remove from the owner of this addon
should I remove?

**Comment by nileshgadgi** [comment]
> > image please remove this variable in kube-state-matrics
> 
> I have not worked on this addons, but I can remove this if gets the permission to remove from the owner of this addon should I remove?

I've worked on it. you can remove it.

**Comment by h1manshu98** [comment]
KEDA outputs are missing in output.tf, please add them.

**Comment by ballalamit** [comment]
> KEDA outputs are missing in output.tf, please add them.

Done

File: override_values.tf [hunk 4]
```diff
@@ -608,3 +608,29 @@ resources:
   EOT
   filename = "${path.module}/override_vales/kube_state_metrics.yaml"
 }
+
+
+resource "local_file" "keda_helm_config" {
+  count    = var.keda && (var.keda_helm_config == null) ? 1 : 0
+  content  = <<EOT
+global:
```
Comment (h1manshu98) [inline override_values.tf:616]: This `global` variable is not present in default keda values,yaml, please fix it.

File: override_values.tf [hunk 5]
```diff
@@ -608,3 +608,29 @@ resources:
...
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: "eks.amazonaws.com/nodegroup"
+            operator: In
+            values:
+            - "critical"
+  ## Using limits and requests
+  resources:
```
Comment (h1manshu98) [inline override_values.tf:627]: there is no `resources:` field present as a direct parent of `limits:`, please fix this.

File: override_values.tf (hunk 4 above)
Comment (ballalamit) [reply override_values.tf]: done

File: override_values.tf (hunk 5 above)
Comment (ballalamit) [reply override_values.tf]: done

File: addons/keda/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,15 @@
+output "service_account" {
```
Comment (h1manshu98) [inline addons/keda/outputs.tf:1]: This can be removed, as we are not using `irsa_config` which creates serviceAccount with mentiond value name.

**Review by h1manshu98 — APPROVED** [review]
LGTM

**Review by anmolnagpal — APPROVED** [review]

**Pull request merged**